### ISSUE TITLE
Add option to not automatically start QUnit

### ISSUE THREAD

**Issue by jamietre** [mmanela/chutzpah#359]
Chutzpah starts QUnit after the require callback resolves for the test file (code gen below):

```
        window.require(amdTestPaths, function () {
            console.log("!!_!! Starting QUnit from inline AMD call...");
            if (!window._Chutzpah_covobj_name) {
                window.QUnit.start();
            }
        });
```

This makes it difficult to ensure that all tests are loaded/run before QUnit starts if using inline requires, or generally, anything that adds a test after some asynchronous activity. I am using a wrapper to abstract dependency resolution in test files, and working around this has proved tricky in the past. The latest update of chutzpah/qunit made this harder (no longer allowed to call `QUnit.stop()` outside of a test). I am continuing to work around this by generating a test before any of the real tests are added that stops QUnit again, but this now causes tests to no longer appear in test explorer. 

It would be great if there was simply an option to have Chutzpah NOT automatically call `QUnit.start()` and just let the end user control it so I didn't have to keep finding hacks to achieve this goal. Alternatively, just respect the "QUnit.config.autostart" property -- if it's already false, then don't start QUnit automatically.

It looks like I could use this `_Chutzpah_covobj_name` global.. which does exactly what I want here.. but this doesn't seem very legit and likely to break in some future release.

What do you think?

**Comment by mmanela** [comment]
I have had requests a long the same line before. In the cases before the issue was people were running explicit nested require calls and Chutzpah was telling the test framework to start way to soon. I have toyed with the idea of having an "explicit" mode where you can configure in the chutzpah.json to tell Chutzpah to not start the test framework until told. And then add some api (like window.Chutzpah.startTestFramework()) to start things. Does this sound reasonable to you?

**Comment by jamietre** [comment]
Thanks, yes - it sounds right. Does the implementation you are thinking of more or less work the way I described above? That is - if I hack the global `_Chutzpah_covobj_name=false` right now, everything is great again. I can call `QUnit.start()` at my leisure - and the test discovery subsytem again works correctly, tests appear in test explorer. 

I don't totally understand the relationship between Chutzpah "starting" and QUnit "starting," but over various iterations of solving my problem I have noticed that I can be in a situation where the tests run correctly in phantomjs, but they don't show up in the test explorer. So ideally the implementation you choose accounts for the different ways the tests can be evaluated and everything will work reliably.
